mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/WaitDelayGeneratorTest.java

```diff
@@ -0,0 +1,58 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.events;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.Duration;
+
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.Test;
+
+class WaitDelayGeneratorTest {
+
+    @Test
+    void generateDelayShouldReturnZeroWhenZeroRetryCount() {
+        WaitDelayGenerator generator = WaitDelayGenerator.of(RetryBackoff.defaultRetryBackoff());
+
+        assertThat(generator.generateDelay(0))
+            .isEqualTo(Duration.ofMillis(0));
+    }
+
+    @Test
+    void generateDelayShouldReturnByRandomInRangeOfExponentialGrowthOfRetryCount() {
```

Comment:
Can you write the same test with...

 - 0 as a jitter factor? I'm then expecting 100, 100, 100...
 - 0 as a first backoff? then there is no wait: 0, 0, 0, 0